File: src/main/java/ru/xpendence/topjavagraduation/entity/Vote.java
package ru.xpendence.topjavagraduation.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "votes")
@Getter
@Setter
public class Vote {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "date")
    private LocalDate date;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "restaurant_id", nullable = false)
    private Restaurant restaurant;

    public static void enrichForUpdate(Vote forUpdate, Vote stored) {
        stored.date = LocalDate.now();
        stored.restaurant = forUpdate.restaurant;
    }
}

File: src/main/kotlin/ru/xpendence/topjavagraduation/entity/Vote.kt
package ru.xpendence.topjavagraduation.entity

import java.time.LocalDate

/**
 * Голос.
 */
data class Vote(
    val id: Long? = null,
    val date: LocalDate,
    val user: User,
    val restaurant: Restaurant,
) {
    companion object {
        @JvmStatic
        fun enrichForUpdate(forUpdate: Vote, stored: Vote): Vote =
            stored.copy(
                date = LocalDate.now(),
                restaurant = forUpdate.restaurant,
            )
    }
}
